examples/solids/qfunctions/finite-strain-neo-hookean-initial-ad.h

```diff
@@ -509,6 +509,26 @@ CEED_QFUNCTION(ElasFSInitialNHdF_AD)(void *ctx, CeedInt Q,
   return 0;
 }
 
+// -----------------------------------------------------------------------------
+// Free tape memory
+// -----------------------------------------------------------------------------
+CEED_QFUNCTION(ElasFSInitialNHFree_AD)(void *ctx, CeedInt Q,
+                                       const CeedScalar *const *in,
+                                       CeedScalar *const *out) {
+  // Inputs
+  void *(*tape) = (void *(*))in[0];
+  // No outputs
+
+  // Quadrature Point Loop
+  CeedPragmaSIMD
+  for (CeedInt i=0; i<Q; i++) {
+    // Free allocated memory for tape
+    free(tape[i]);
```

Comment:
This seems to be an invalid pointer. 
```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff3feb859 in __GI_abort () at abort.c:79
#2  0x00007ffff40563ee in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff4180285 "%s\n") at ../sysdeps/posix/libc_fatal.c:155
#3  0x00007ffff405e47c in malloc_printerr (str=str@entry=0x7ffff417e4ae "free(): invalid pointer") at malloc.c:5347
#4  0x00007ffff405fcac in _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:4173
#5  0x000000000042e319 in ElasFSInitialNHFree_AD (ctx=<optimized out>, Q=216, in=<optimized out>, out=<optimized out>)
    at problems/../qfunctions/finite-strain-neo-hookean-initial-ad.h:526
#6  0x00007ffff529839f in CeedQFunctionApply_Ref (qf=0xfc3900, Q=216, U=0x24f89d0, V=0x24f8a60) at /home/leila/git/libCEED/backends/ref/ceed-ref-qfunction.c:54
```
I am either missing something or this is not the right approach to free the allocated memory for tape. @jedbrown @jeremylt , your insight would be appreciated!